### PR TITLE
test: add wait for block to tss migration test

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description

The change might improve the success frequency of the TSS-migration-test on the CI .

The pr adds a loop to make sure the new TSS is set before trying to migrate funds. 

# How Has This Been Tested?

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Summary by CodeRabbit

- **New Features**
	- Enhanced TSS generation monitoring capabilities in the end-to-end testing framework.
	- Introduced methods to ensure TSS generation is completed before proceeding with Bitcoin fund migration.
  
- **Bug Fixes**
	- Improved error handling for TSS generation discrepancies.

### CODE REVIEW

File: e2e/runner/zeta.go [hunk 1]
```diff
@@ -31,6 +34,30 @@ func (r *E2ERunner) WaitForBlocks(n int64) {
 	err = retry.DoWithBackoff(call, boWithMaxRetries)
 	require.NoError(r, err, "failed to wait for %d blocks", n)
 }
+
+// WaitForTssGeneration waits for a specific number of TSS to be generated
+// The parameter n is the number of TSS to wait for
+func (r *E2ERunner) WaitForTssGeneration(n int64) {
+	call := func() error {
```
Comment (swift1337) [inline e2e/runner/zeta.go:41]: consider using `pkg/retry`

**Review by swift1337 — APPROVED** [review]

File: e2e/runner/zeta.go [hunk 2]
```diff
@@ -31,6 +34,30 @@ func (r *E2ERunner) WaitForBlocks(n int64) {
 	err = retry.DoWithBackoff(call, boWithMaxRetries)
 	require.NoError(r, err, "failed to wait for %d blocks", n)
 }
+
+// WaitForTssGeneration waits for a specific number of TSS to be generated
+// The parameter n is the number of TSS to wait for
+func (r *E2ERunner) WaitForTssGeneration(n int64) {
```
Comment (lumtis) [inline e2e/runner/zeta.go:40]:
```suggestion
func (r *E2ERunner) WaitForTSSGeneration(n int64) {
```

File: e2e/runner/zeta.go [hunk 3]
```diff
@@ -31,6 +34,30 @@ func (r *E2ERunner) WaitForBlocks(n int64) {
...
+	call := func() error {
+		return retry.Retry(r.waitForTssGeneration(n))
+	}
+	bo := backoff.NewConstantBackOff(time.Second * 5)
+	boWithMaxRetries := backoff.WithMaxRetries(bo, 10)
+	err := retry.DoWithBackoff(call, boWithMaxRetries)
+	require.NoError(r, err, "failed to wait for %d tss generation", n)
+}
+
+func (r *E2ERunner) waitForTssGeneration(n int64) error {
```
Comment (lumtis) [inline e2e/runner/zeta.go:50]: Name misleading, we don't wait in this function, we directly fetch and check list

File: e2e/runner/zeta.go (hunk 2 above)
Comment (lumtis) [inline e2e/runner/zeta.go:40]:
```suggestion
func (r *E2ERunner) WaitForTssGeneration(tssNumber int64) {
```

**Review by lumtis — CHANGES_REQUESTED** [review]

File: e2e/runner/zeta.go [hunk 4]
```diff
@@ -31,6 +34,30 @@ func (r *E2ERunner) WaitForBlocks(n int64) {
...
+	call := func() error {
+		return retry.Retry(r.checkNumberOfTssGenerated(tssNumber))
+	}
+	bo := backoff.NewConstantBackOff(time.Second * 5)
+	boWithMaxRetries := backoff.WithMaxRetries(bo, 10)
+	err := retry.DoWithBackoff(call, boWithMaxRetries)
+	require.NoError(r, err, "failed to wait for %d tss generation", tssNumber)
+}
+
+func (r *E2ERunner) checkNumberOfTssGenerated(tssNumber int64) error {
```
Comment (lumtis) [inline e2e/runner/zeta.go:50]:
```suggestion
func (r *E2ERunner) checkNumberOfTSSGenerated(tssNumber int64) error {
```
We really need to always remember using fully uppercase for acronym, this is small detail but still help for code clarity

**Review by lumtis — APPROVED** [review]

**Pull request merged**